### PR TITLE
[WebNN EP] Support ConvTranspose for TFLite backend

### PR DESCRIPTION
**Pull request by Honry**
### Description
Chromium supports ConvTranspose for TFLite in
https://chromium-review.googlesource.com/c/chromium/src/+/5635194

With constraint that only default dilations and groups are supported.

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @guschmue, PTAL, thanks!

**Comment by Honry** [comment]
@guschmue, @fdwr, looks like the CI failure is irrelevant to my change. Could you pls. help retrigger them? Thanks!

**Review by fdwr — DISMISSED** [review]
Is this a regression relative to XNNPack then?

**Comment by Honry** [comment]
> Is this a regression relative to XNNPack then?

Umm, not sure, @fujunwei may help answer this.

File: onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc [hunk 1]
```diff
@@ -446,6 +446,22 @@ bool ConvOpBuilder::HasSupportedInputsImpl(const Node& node, const WebnnDeviceTy
     return false;
   }
 
+  // WebNN CPU backend (TFLite) only supports default dilations and group.
+  // https://source.chromium.org/chromium/chromium/src/+/main:services/webnn/tflite/graph_builder_tflite.cc;l=1040
+  if (device_type == WebnnDeviceType::CPU && op_type == "ConvTranspose") {
+    NodeAttrHelper helper(node);
+    const auto dilations = helper.Get("dilations", std::vector<int64_t>{1, 1});
+    const auto group = helper.Get("group", 1);
+    if (dilations[0] !=1 || (dilations.size() > 1 && dilations[1] != 1)) {
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc:455]:
```suggestion
    if (dilations[0] != 1 || (dilations.size() > 1 && dilations[1] != 1)) {
```
Linter requires it.

**Review by fdwr — DISMISSED** [review]

**Comment by fdwr** [comment]
@honry Consider merging from main again, which might help the failed tests (also, I can't approve because I fixed the lint issue, and you're not allowed to approve when you were the last pusher).

**Review by fdwr — APPROVED** [review]

**Review by guschmue — APPROVED** [review]

**Comment by fdwr** [comment]
@honry So I restarted the ORT Web CI pipeline thinking it was flakiness, but it failed again. However, I'm not seeing a clear error o_O.

**Comment by Honry** [comment]
> @Honry So I restarted the ORT Web CI pipeline thinking it was flakiness, but it failed again. However, I'm not seeing a clear error o_O.

Umm... What can I do? Trying to merge the latest main again?

**Comment by fdwr** [comment]
The ORT Web CI pipeline issue appears unrelated to this change:

> gmake: *** [Makefile:720: onnxruntime_webassembly] Error 2
> ...
> [onnxruntime-extensions] Loading onnxruntime-extensions from: FetchContent

Continuing merge.

File: onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc [hunk 2]
```diff
@@ -427,6 +427,22 @@ bool ConvOpBuilder::HasSupportedInputsImpl(const Node& node, const WebnnDeviceTy
     return false;
   }
 
+  // WebNN CPU backend (TFLite) only supports default dilations and group.
+  // https://source.chromium.org/chromium/chromium/src/+/main:services/webnn/tflite/graph_builder_tflite.cc;l=1040
+  if (device_type == WebnnDeviceType::CPU && op_type == "ConvTranspose") {
```
Comment (satyajandhyala) [inline onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc:432]: device_type variable is undefined

Comment (fdwr) [reply onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc:432]: Arg, I would appreciate less misleading output from the build process.

Comment (fdwr) [reply onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc:432]: Follow up fix here: https://github.com/microsoft/onnxruntime/pull/21569

**Pull request merged**